src/object/schedcontrol.c

```diff
@@ -28,9 +28,11 @@ static exception_t invokeSchedControl_Configure(sched_context_t *target, word_t
 #ifdef ENABLE_SMP_SUPPORT
             if (target->scCore == getCurrentCPUIndex()) {
 #endif /* ENABLE_SMP_SUPPORT */
-                if (checkBudget()) {
-                    commitTime();
-                }
+                /* This could potentially mutate state but if it returns
+                 * true no state was modified, thus removing it should
+                 * be the same. */
+                assert(checkBudget());
```

Comment:
Have you run all of the multicore tests and benchmarks on this? I added this when doing multicore benchmarks as I saw problems. Could be redundant, could also be only relevant on multicore.

Reply:
I will have to look into that.

Reply:
I looked back through the history for this and when the code was added, checkBudget was used to test whether to do a commitTime or chargeBudget based on if the SC had used up its budget or not and also applied to SCs currently running on separate cores:
```
        if (NODE_STATE_ON_CORE(ksCurSC, target->scCore) == target) {
            ticks_t capacity = refill_capacity(target, NODE_STATE_ON_CORE(ksConsumed, target->scCore));
            if (checkBudget()) {
                commitTime();
            } else {
                chargeBudget(capacity, NODE_STATE_ON_CORE(ksConsumed, target->scCore), false);
            }
        }
```
Later on the chargeBudget was moved into a branch that would only happen if the SC was on a different core:
```
        if (NODE_STATE_ON_CORE(ksCurSC, target->scCore) == target) {
#ifdef ENABLE_SMP_SUPPORT
            if (target->scCore == getCurrentCPUIndex()) {
#endif /* ENABLE_SMP_SUPPORT */
                if (checkBudget()) {
                    commitTime();
                }
#ifdef ENABLE_SMP_SUPPORT
            } else {
                /* if its a remote core, manually charge the budget */
                ticks_t capacity = refill_capacity(target, NODE_STATE_ON_CORE(ksConsumed, target->scCore));

                chargeBudget(capacity, NODE_STATE_ON_CORE(ksConsumed, target->scCore), false, target->scCore, false);
                doReschedule(target->scCore);
            }
#endif /* ENABLE_SMP_SUPPORT */
        }
```
Now that the other core is treated in a different condition, the `checkBudget()` in the case where the SC is on the current core is known to always return true due to the `checkBudget()` performed at the start of `handleSyscall()` and so treating it as an assert here now seems to be ok.

Reply:
In that case this commit should also be fine